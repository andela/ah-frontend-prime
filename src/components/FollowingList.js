import React from "react";

const FollowingList = props => {
  const { following } = props;
  return (
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title">My Following</h5>
          <button
            type="button"
            className="close"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          {following && following.length > 0
            ? following.map(item => (
                <div
                  className="row"
                  id="following"
                  key={item.username}
                  onClick={() => {
                    document.location.href = `/profile/${item.username}`;
                  }}
                  key={item.username}
                >
                  <button className="col col-lg-3" style={{ border: "none" }}>
                    <img
                      src={
                        item.image
                          ? `${item.image}`
                          : "https://cdn.pixabay.com/photo/2013/07/13/10/07/man-156584__340.png"
                      }
                      alt="profile"
                      className="profile-page"
                    />
                  </button>
                  <div className="col col-lg-9">
                    <h5>{item.username}</h5>
                    <p>{item.full_name}</p>
                  </div>
                  <br />
                </div>
              ))
            : "No one is on the following List"}
        </div>
        <div class="modal-footer" />
      </div>
    </div>
  );
};

export default FollowingList;
