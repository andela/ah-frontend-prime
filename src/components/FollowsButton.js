import React from "react";

const FollowsButton = props => {
  const { isfollowing, unfollowUser, followUser } = props;
  return (
    <button
      onClick={
        isfollowing
          ? unfollowUser.bind(this, sessionStorage.getItem("userview_name"))
          : followUser.bind(this, sessionStorage.getItem("userview_name"))
      }
      className="btn btn-outline-success"
    >
      {isfollowing ? "Unfollow" : "Follow"}
    </button>
  );
};

export default FollowsButton;
