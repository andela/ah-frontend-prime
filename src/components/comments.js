import React, { Component } from "react";
import moment from "moment";
import { articleComments, fetchComments } from "../actions/commentsActions";
import { connect } from "react-redux";
import "../styles/comments.scss";


export class Comments extends Component {
  constructor(props) {
    super(props);
    this.state = {
      comments: ""
    };
  }

  componentDidMount() {
    const { fetchComments, articleSlug } = this.props;
    fetchComments(articleSlug);
  }

  handleOnChange = event => {
    event.preventDefault();
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleOnSubmit = event => {
    event.preventDefault();
    const { articleSlug, articleComments, fetchComments } = this.props;
    const { comments } = this.state;
    const data = {
      body: comments
    };
    articleComments(data, articleSlug);
    fetchComments(articleSlug);
  };

  render() {
    const { commentsMessage } = this.props;
    const comment = commentsMessage.map(message => (
      <section className="comments" key={message.id}>
        <article className="comment">
          <a className="comment-img" href="#non">
            <img src={message.author.image} alt="" width="50" height="50" />
          </a>
          <div className="comment-body">
            <div className="text">
              <p>{message.body}</p>
            </div>
            <p className="attribution">
              by <a href={`/profile/${message.author.username}`}>{message.author.username}</a> at {moment(message.createdAt).format(
                          "h:mm:ss a [on] MMMM Do YYYY.")}
                        
            </p>
          </div>
        </article>
      </section>
    ));
    return (
      <div className="row">
        <div className="col col-lg-1" />
        <div className="col col-lg-10">
          <div className="row">
            <div className="col col-lg-1" />
            <div className="col col-lg-11">
              <form
                method="POST"
                onSubmit={this.handleOnSubmit}
                className="comment-forms"
              >
                <textarea
                  className="form-control"
                  name="comments"
                  value={this.state.comments}
                  onChange={this.handleOnChange}
                >
                  comments here
                </textarea>
                <button
                  type="submit"
                  className="btn btn-comment"
                  style={{ marginTop: "10px" }}
                >
                  comment
                </button>
              </form>
            </div>
          </div>
          {comment}
        </div>
        <div className="col col-lg-1" />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  commentsSuccess: state.comments.success,
  commentsErrors: state.comments.errors,
  commentsMessage: state.getComments.comments
});
export default connect(
  mapStateToProps,
  { articleComments, fetchComments }
)(Comments);
