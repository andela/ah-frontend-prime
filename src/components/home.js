import React, { Component } from "react";
import { connect } from "react-redux";
import "../styles/app.scss";
import moment from "moment";
import { Link } from "react-router-dom";
import { getArticlesAction } from "../actions/getArticles";

export class Home extends Component {
  constructor(props) {
    super(props);
    this.fetchNext = this.fetchNext.bind(this);
    this.fetchPrevious = this.fetchPrevious.bind(this);
  }
  componentDidMount() {
    const { getArticlesAction } = this.props;
    getArticlesAction();
  }
  fetchNext() {
    const { next, getArticlesAction } = this.props;
    getArticlesAction(next);
  }
  fetchPrevious() {
    const { previous, getArticlesAction } = this.props;
    getArticlesAction(previous);
  }
  render() {
    const { articles } = this.props;
    const articleList = articles.length ? (
      articles.map(article => {
        return (
          <div className="article" key={article.slug}>
            <div className="inner">
              <Link
                className="links"
                to={{
                  pathname: "/article/" + article.slug
                }}
              >
                <div className="article-card">
                  <h3>{article.title.substring(0, 30)} ...</h3>
                  <p>
                    <b>{article.description.substring(0, 60)} ...</b>
                  </p>
                  <span>
                    <b>Author: {article.author.username}</b>
                  </span>
                  <br />

                  <span className="card-date">
                    Created at{" "}
                    {moment(article.createdAt).format(
                      "h:mm:ss a [on] MMMM Do YYYY."
                    )}
                  </span>
                  <br />
                  <p>
                    <b>{article.reading_time}</b>
                  </p>
                </div>

                <div>
                  <img
                    src={
                      article.image
                        ? article.image
                        : "https://previews.123rf.com/images/twindesigner/twindesigner1708/twin" +
                          "designer170800135/84202763-ah-brush-letter-logo-design-with-black-" +
                          "circle-creative-brushed-letters-icon-logo-.jpg"
                    }
                    className="img-tile"
                  />
                </div>
              </Link>
            </div>
          </div>
        );
      })
    ) : (
      <div>No Articles in Authors Haven Yet</div>
    );

    return (
      <div className="big-container">
        <div className="articles-top">
          <div className="best-article">
            <div className="inner">
              <img
                src={
                  "http://allpicts.in/download/22916/2018/03/Natural_Images_HD_1" +
                  "080p_Download_with_Waterfall_in_Tollymore_Forest_Park-1440x900.jpg/"
                }
                className="img-top"
              />
              <div className="inner">
                <div className="inner">
                  <div className="best-article-card">
                    <h3>
                      Facebook's Plan to Fuse its Messaging Apps Is Not about
                      Your Privacy Just follow the Money
                    </h3>
                    <span>
                      <b>Author: Yasha Levin in OneZero</b>
                    </span>
                    <br />
                    <span className="card-date">Created a few hours ago</span>
                  </div>
                </div>
              </div>
            </div>
            <div />
          </div>
          <div className="top-articles">{articleList}</div>
          <div className="top-articles">{articleList}</div>
        </div>

        <div className="articles">
          <div className="heading">
            <h3>Featured Articles</h3>
          </div>
          <hr />
          {articleList}
        </div>
        <div className="articles1">
          <div className="heading">
            <h3>Popular Articles</h3>
          </div>
          <hr />
          {articleList}
        </div>
        <div>
          <button
            className="btn btn-outline-success btn eian"
            onClick={() => this.fetchPrevious()}
          >
            Previous
          </button>{" "}
          <button
            className="btn btn-outline-success btn aken"
            onClick={() => this.fetchNext()}
          >
            Next
          </button>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  articles: state.getArticlesReducer.articles,
  next: state.getArticlesReducer.next,
  previous: state.getArticlesReducer.previous
});

export default connect(
  mapStateToProps,
  { getArticlesAction }
)(Home);
