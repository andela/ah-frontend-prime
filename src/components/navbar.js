import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import "../styles/navbar.scss";
import BellIcon from "../styles/images/icons8-bell-100.png";
import SideDrawer from "./sideDrawer";

export class NavBarComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      navBarDrawer: false
    };
  }

  toggleDrawer = event => {
    event.preventDefault();
    this.setState(prevState => {
      return { navBarDrawer: !prevState.navBarDrawer };
    });
  };

  render() {
    let sideDrawer;
    if (this.state.navBarDrawer) {
      sideDrawer = <SideDrawer />;
    }

    const { isAuthenticated } = this.props;
    return (
      <div style={{ height: "100%" }}>
        <nav className="navbar navbar-expand-sm ">
          <ul className="navbar-nav">
            <li className="nav-item">
              <a className="nav-link" href="#">
                <h4>
                  <i
                    className="fas fa-bars"
                    style={{ marginRight: "20px" }}
                    onClick={this.toggleDrawer}
                    id="nav-icon"
                  />
                  Author's Haven
                </h4>
              </a>
            </li>
          </ul>
          {isAuthenticated ? (
            <ul className="nav navbar-nav ml-auto">
              <li className="nav-item active">
                <div className="form-group has-search">
                  <span
                    className="fa fa-search form-control-feedback"
                    id="search-icon"
                  />
                  <form className="form-inline my-5 my-lg-0">
                    <input
                      className="form-control mr-sm-2 form-control-search"
                      type="search"
                      aria-label="Search"
                    />
                  </form>
                </div>
              </li>
              <li className="nav-item">
                <img src={BellIcon} id="notification-icon" />
              </li>
              <li className="nav-item" />
              <li className="nav-item">
                <div>
                  <img src="" className="profile-section" />
                </div>
              </li>
            </ul>
          ) : (
            <ul className="nav navbar-nav ml-auto">
              <li className="nav-item">
                <Link className="nav-link" to="/login">
                  Sign In
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link get-started " to="/register">
                  Get Started
                </Link>
              </li>
            </ul>
          )}
        </nav>
        {sideDrawer}
      </div>
    );
  }
}

NavBarComponent.proptypes = {
  isAuthenticated: PropTypes.Boolean
};
const mapStateToProps = state => ({
  isAuthenticated: state.auth_login.loginSuccess
});
export default connect(mapStateToProps)(NavBarComponent);
