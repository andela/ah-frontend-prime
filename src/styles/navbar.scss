.has-search .form-control {
  padding-left: 2.375rem;
}
.dropbtn {
  background-color: #4caf50;
  cursor: pointer;
}


.dropdown {
  position: relative;
  display: inline-block;
}

.dropdown-content {
  display: none;
  position: absolute;
  right: 0px;
  color: blue;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
}

.dropdown-content {
  background: white;
  .navlink {
    color: black;
    display: block;
    padding: 10px;
  }
}

.dropdown-content a:hover {
  background-color: #003153;
  color: white;
}


.dropdown:hover .dropdown-content {
  display: block;
}


.dropdown:hover .dropbtn {
  background-color: #3e8e41;
}
.profile-img {
  border: 1px solid inherit;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  background: inherit;
}

.has-search .form-control-feedback {
  position: absolute;
  z-index: 2;
  display: block;
  width: 6.375rem;
  height: 2.375rem;
  line-height: 2.375rem;
  text-align: center;
  pointer-events: none;
  color: #aaa;
  background: none;
}

#search-icon {
  color: white;
  margin-top: 3px;
  margin-left: 150px;
}

#search-icon:focus {
  display: none;
}

form {
  input.form-control-search {
    background: white;
    border-radius: 20px;
    margin-top: 3px;
    background: none;
    border: 1px solid white;
  }

  input.form-control-search:focus {
    background: none;
    border: 1px solid white;
    color: white;
  }
}

.navbar {
  background: #003153;
  color: white;

  .nav-item {
    .nav-link {
      color: white;
      padding-top: 10px;
    }

    #notification-icon {
      color: white;
      width: 40px;
      margin-left: 30px;
      margin-right: 20px;
    }

    .profile-section {
      border: 1px solid white;
      width: 40px;
      height: 40px;
      border-radius: 50%;
      margin-right: 30px;
    }

    .get-started {
      border: 1px solid white;
      border-radius: 5px;
      margin-left: 30px;
    }
    .log-out {
      border: 1px solid white;
      border-radius: 5px;
      margin-left: 30px;
      color: white;
      background-color: inherit;
    }
  }
}

@media (max-width: 360px) {
  .nav {
    .nav-item {
      .nav-link {
        font-size: 12px;

        h4 {
          font-size: 15px;
        }
      }
    }
  }
}

@media (max-width: 567px) {
  .nav {
    .nav-item {
      display: none;
    }
  }
}

@media (max-width: 1600px) {
  #nav-icon {
    display: none;
  }
}

@media (max-width: 995px) {
  .nav {
    display: inline;
  }
}

@media (max-width: 534px) {
  #nav-icon {
    display: inline;
    margin-right: 100px;
  }
}
