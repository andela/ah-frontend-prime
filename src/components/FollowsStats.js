import React from "react";
import FollowingList from "./FollowingList";
import FollowersList from "./FollowersList";

const FollowsStats = props => {
  const { following_count, followers_count, followers, following } = props;
  return (
    <div className="row">
      <div
        className="modal fade"
        id="exampleModal"
        tabIndex={-1}
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <FollowingList following={following} />
      </div>
      <div
        className="modal fade"
        id="exampleModal1"
        tabIndex={-1}
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <FollowersList followers={followers} />
      </div>
      <a
        className="col-md-4"
        data-toggle="modal"
        data-target="#exampleModal1"
        href="#"
      >
        {followers_count === 1 ? "Follower" : "Followers"}
        {" : "}
        <button className="stats">
          <strong>{followers_count ? followers_count : 0} </strong>
        </button>
      </a>
      <a
        className="col-md-4"
        data-toggle="modal"
        data-target="#exampleModal"
        href="#"
      >
        Following{" : "}
        <button className="stats">
          <strong>{following_count ? following_count : "0"} </strong>
        </button>
      </a>
    </div>
  );
};

export default FollowsStats;
