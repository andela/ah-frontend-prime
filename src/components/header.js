import React from "react";
import PropTypes from "prop-types";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import InputBase from "@material-ui/core/InputBase";
import { fade } from "@material-ui/core/styles/colorManipulator";
import { withStyles } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";

const styles = theme => ({
  root: {
    width: "100%"
  },
  grow: {
    flexGrow: 1
  },
  menu: {
    marginLeft: -12,
    marginRight: 20
  },
  title: {
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block"
    }
  },
  inputRoot: {
    color: "inherit",
    width: "100%"
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 9,
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      width: 118,
      "&:focus": {
        width: 180
      }
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 8,
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  },
  bar: {
    backgroundColor: "#003153",
    boxShadow: "none"
  },
  search: {
    position: "relative",
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.18),
    "&:hover": {
      backgroundColor: fade(theme.palette.common.white, 0.29)
    },
    marginLeft: 0,
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      marginLeft: theme.spacing.unit,
      width: "auto"
    }
  }
});

function TopAppBar(props) {
  const { classes } = props;
  return (
    <div className={classes.root}>
      <AppBar position="static" className={classes.bar}>
        <Toolbar>
          <Typography
            className={classes.title}
            variant="h6"
            color="inherit"
            noWrap
          >
            Authors Haven
          </Typography>
          <div className={classes.grow} />

          <div className={classes.search}>
            <div className={classes.searchIcon}>
              <SearchIcon />
            </div>
            <InputBase
              placeholder="Searchâ€¦"
              classes={{
                root: classes.inputRoot,
                input: classes.inputInput
              }}
            />
          </div>
          <div className="nav-div">
            <a href="login" className="nav-links">
              Sign In
            </a>
            &nbsp;&nbsp; | &nbsp;&nbsp;
            <a href="register" className="nav-links">
              Get Started
            </a>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
}

TopAppBar.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(TopAppBar);
