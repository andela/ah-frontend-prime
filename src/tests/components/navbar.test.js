import React from "react";
import { NavBarComponent } from "../../components/navbar";
import { shallow } from "enzyme";

let wrapper;
let instance;
const props = {
  navBarDrawer: false,
  toggleDrawer: jest.fn(),
  logout: jest.fn()
};
describe(" navbar component", () => {
  beforeEach(() => {
    wrapper = shallow(<NavBarComponent {...props} />);
    instance = wrapper.instance();
  });
  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should test toggling of the navbar", () => {
    const e = {
      preventDefault: jest.fn()
    };
    props.toggleDrawer(e);
    wrapper.setState({ navBarDrawer: true });
    expect(props.navBarDrawer).toEqual(false);
  });
  it("should test handle toggle method", () => {
    const e = {
      preventDefault: jest.fn()
    };
    instance.toggleDrawer(e);
    expect(props.navBarDrawer).toEqual(false);
  });
  it("should load other nav elements when a user is not authenticated", () => {
    wrapper.setProps({
      isAuthenticated: '<ul className="nav navbar-nav ml-auto"> </ul>'
    });
    expect(wrapper).toMatchSnapshot();
  });
  it("should handle onclick event", () => {
    instance.logout();
    expect(props.logout).toBeCalledTimes(0);
  });
});
