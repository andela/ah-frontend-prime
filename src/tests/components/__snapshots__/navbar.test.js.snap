// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` navbar component should load other nav elements when a user is not authenticated 1`] = `ShallowWrapper {}`;

exports[` navbar component should render without crashing 1`] = `ShallowWrapper {}`;
